chain/index/interface.go

```diff
@@ -24,18 +24,19 @@ type MsgInfo struct {
 
 // MsgIndex is the interface to the message index
 type MsgIndex interface {
-	// GetMsgInfo retrieves the message metadata through the index.
+	// GetMsgInfo looks up the message in index and retrieves its metadata and execution
+	// tipset cid.
 	// The lookup is done using the onchain message Cid; that is the signed message Cid
 	// for SECP messages and unsigned message Cid for BLS messages.
-	GetMsgInfo(ctx context.Context, m cid.Cid) (MsgInfo, error)
+	GetMsgInfo(ctx context.Context, mCid cid.Cid) (MsgInfo, cid.Cid, error)
```

Comment:
If I'm not wrong, the `MsgInfo` isn't actually serialized and stored anywhere, which means we can break the shape of the struct without any migration or other problems. In that case, consider just adding the xTsCid to `MsgInfo`, instead of making it a separate returned value?